### DISCUSSION TITLE
Having a hard time understanding how to add a custom interface as a member to one of my models

### DISCUSSION THREAD

**Discussion by Tsury** [xaviergonz/mobx-keystone#537] [Q&A] [answered]
I'm new to `mobx-keystone` and didn't manage to find an answer to my (basic) question:

Suppose I have this model:

```
class MyModel extends Model({
  someNumber: tProp(types.number, 0).withSetter()
}) {}
```

Suppose I have this interface:

```
interface MyInterface {
  someData: string;
  someOtherData: boolean;
}
```

How can I add a new member to my model, called `someMyInterface` (I also need it to be optional but I digress)?

Only way I managed was either:
`someMyInterface: tProp(types.unchecked<PlayerHeroSwap>()).withSetter(),`
Which bails me out of the type check,

Or:
`someMyInterface: tProp(types.object(() => ({ someData: types.string, someOtherData: types.boolean })))`
Which bypasses the interface altogether, which is undesired.

**Comment by xaviergonz** [comment, marked as answer]
The first way you used is valid.
A faster way to do it skipping the *runtime* type checking is:
```
someMyInterface: prop<MyInterface>().withSetter()
```
if you want runtime type checking then an interface is not enough, you need to tell the runtime how it should be type checked (which is what types.object is actually doing)

**Reply by Tsury** [reply]
Got it, thank you for the clarification!
